### DISCUSSION TITLE
Noah 2000 always in the internet?

### DISCUSSION THREAD

**Discussion by MHaneu** [mtrossbach/noah-mqtt#47] [General]
Hello Markus,
thank you for the great project.
Is there a way / possibility  to communicate directly with the storage without the Noah 2000 is always in internet? 
My idea is to control the output via home assistant to waste no energie in the grid. To use the growatt server have two big dissadvantages:
- reaction time is longer than a direct connection
- connection to the server can be lost, means control is lost
Do you see any possibility for a direct connection?
Or do I make a misstake by the setup? (Because if I disconect the NOAH from the internet, I will not recieve any data.)

**Comment by xdatravelbug** [comment]
Your understanding is correct: There is currently no way connecting to the Noah2000 without cloud - internet is needed!

There is a community for the Growatt inverter, who use a custom Wifi connection via, e.g., ESP32. As no-one seemed having opened up the Noah2000, we do not know yet if the same could be applied here (in case that the Wifi-module is internally connected via USB/serial). Another possibility would be, re-flashing the internal Wifi via a FW-update (similar to what is done in the Roborock community). If one of these is the case everything can be handled offline.

**Reply by les-les** [reply]
might it be possible connecting the noah tonthis esp32 wifi device ?

**Reply by xdatravelbug** [reply]
In general I would say 'yes', but - again - I did not see an open Noah2000 yet. Maybe because of the warranty?! Once we have that information how the thing looks on the inside we can go the next step. I may have access to an x-ray machine at work but the battery will most likely not fit in :-(

**Reply by les-les** [reply]
Some gernan youtuber have opened the noah 
I'll try to reach out to one of them maybe the can share some Details with us here 

Sounds like a fun project
